syntax = "proto3";

package backup_primary;
service BackupPrimarygRPC {
    rpc RestoreData (stream RestoreDataReq) returns (RestoreDataReply) {}
    rpc PrimaryHeartbeat (PrimaryHeatbeatReq) returns (PrimaryHeatbeatReply){}
}
  
// Restore Write data
message RestoreDataReq {
    int32  offset = 1;
    bytes  buf = 2;
}

message RestoreDataReply {
    int32 err = 1;
}

// Primary Heatbeat to Backup
message PrimaryHeatbeatReq {
    int32  beat = 1;
}

message PrimaryHeatbeatReply {
    int32 beat = 1;
}
